packages/runtime/container-runtime/src/dataStoreContext.ts

```diff
@@ -812,6 +818,21 @@ export abstract class FluidDataStoreContext
 		checkTombstone = true,
 		safeTelemetryProps: ITelemetryProperties = {},
 	) {
+		if (this.deleted) {
+			const messageString = `Context is deleted! Call site [${callSite}]`;
+			const error = new DataCorruptionError(messageString, safeTelemetryProps);
+			this.mc.logger.sendErrorEvent(
+				{
+					eventName: "GC_Deleted_DataStore_Changed",
+					isSummarizerClient: this.clientDetails.type === summarizerClientType,
+					callSite,
+				},
+				error,
+			);
+
+			throw error;
+		}
+
 		if (this._disposed) {
```

Comment:
What's your thought process for putting the `deleted` check first? I assumed the `_disposed` check would be first but I can see it both ways I guess

Reply:
A context could be disposed, and not deleted, and disposed and deleted. If you're touching it while its deleted when disposed and deleted, we would still want to prevent any changes. Though there's not that much of a difference either way.